src/status_im/contexts/chat/home/add_new_contact/events.cljs

```diff
@@ -163,21 +153,35 @@
                              (string/includes? (:message err) "no such host")))
                     {:state :invalid :msg :t/lost-connection}
                     :else {:state :invalid})]
-        {:db (assoc db :contacts/new-identity (merge contact state))}))))
+        (merge {:db (assoc db :contacts/new-identity (merge contact state))}
+               (when failure-fn
+                 (failure-fn)))))))
 
-(rf/defn clear-new-identity
-  {:events [:contacts/clear-new-identity :contacts/new-chat-focus]}
-  [{:keys [db]}]
-  {:db (dissoc db :contacts/new-identity)})
+(re-frame/reg-event-fx :contacts/set-new-identity-error set-new-identity-error)
+
+(defn build-contact
+  [{:keys [pubkey ens success-fn]}]
+  {:json-rpc/call [{:method      "wakuext_buildContact"
+                    :params      [{:publicKey pubkey
+                                   :ENSName   ens}]
+                    :js-response true
+                    :on-success  #(re-frame/dispatch [:contacts/build-contact-success
+                                                      {:pubkey     pubkey
+                                                       :contact    (data-store.contacts/<-rpc-js %)
+                                                       :success-fn success-fn}])}]})
 
-(rf/defn qr-code-scanned
-  {:events [:contacts/qr-code-scanned]}
-  [{:keys [db] :as cofx} scanned]
-  (rf/merge cofx
-            (set-new-identity scanned scanned)
-            (navigation/navigate-back)))
+(re-frame/reg-event-fx :contacts/build-contact (fn [_ [contact-data]] (build-contact contact-data)))
```

Comment:
why not just add `build-contact` functionality directly here?

Reply:
Fixed!